### PR TITLE
Workspace folders API

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
Plugin API to get information about workspace folder.

Signed-off-by: Vitaliy Gulyy <vgulyy@redhat.com>

### CODE REVIEW

File: packages/plugin/src/theia.d.ts [hunk 1]
```diff
@@ -2101,6 +2141,37 @@ declare module '@theia/plugin' {
...
     export namespace workspace {
+        /**
+         * ~~The folder that is open in the editor. `undefined` when no folder
+         * has been opened.~~
+         *
+         * @deprecated Use [`workspaceFolders`](#workspace.workspaceFolders) instead.
+         *
+         * @readonly
+         */
+        export let rootPath: string | undefined;
```
Comment (evidolob) [inline packages/plugin/src/theia.d.ts]: This is deprecated field in vscode workspace api, can you remove it?

File: packages/plugin-ext/src/plugin/workspace.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import { Path } from "@theia/core/lib/common/path";
+
+export class WorkspaceExtImpl implements WorkspaceExt {
+
+    private workspaceFoldersChangedEmitter = new Emitter<WorkspaceFoldersChangeEvent>();
+    public readonly onDidChangeWorkspaceFolders: Event<WorkspaceFoldersChangeEvent> = this.workspaceFoldersChangedEmitter.event;
+
+    private folders: WorkspaceFolder[] | undefined;
+
+    constructor(rpc: RPCProtocol) {
```
Comment (mmorhun) [inline packages/plugin-ext/src/plugin/workspace.ts]: Why do you need rpc here?

File: packages/plugin/src/theia.d.ts (hunk 1 above)
Comment (vitaliy-guliy) [reply packages/plugin/src/theia.d.ts]: Done

File: packages/plugin-ext/src/plugin/workspace.ts (hunk 2 above)
Comment (vitaliy-guliy) [reply packages/plugin-ext/src/plugin/workspace.ts]: Removed. Thanks

File: packages/plugin-ext/src/main/browser/main-context.ts [hunk 3]
```diff
@@ -34,6 +37,12 @@ export function setUpPluginApi(rpc: RPCProtocol, container: interfaces.Container
     // @ts-ignore
     const windowStateMain = new WindowStateMain(rpc);
 
+    const workspaceService: WorkspaceService = container.get(WorkspaceService);
+
+    // tslint:disable-next-line:no-unused-variable
+    // @ts-ignore
+    const workspaceMain = new WorkspaceMain(rpc, workspaceService);
```
Comment (benoitf) [inline packages/plugin-ext/src/main/browser/main-context.ts]: do we need to declare the variable ? if not used

Comment (vitaliy-guliy) [reply packages/plugin-ext/src/main/browser/main-context.ts]: Removed. Thanks.

**Pull request merged**